File: ios/FamilyApp/Features/Home/HomeScreen.swift
// Home dashboard — the iOS twin of HomeScreen.kt: greeting header with avatar,
// gradient family card / no-family banner, glanceable summary cards, feature tile grid.
import NukeUI
import SwiftUI

private struct Feature: Identifiable {
    let title: String
    let subtitle: String
    let systemImage: String
    let accent: FeatureAccent
    /// nil = Calendar, which is a bottom tab rather than a pushed route.
    let route: Route?

    var id: String {
        title
    }
}

private let features: [Feature] = [
    Feature(
        title: "Shopping",
        subtitle: "Shared lists",
        systemImage: "cart.fill",
        accent: .shopping,
        route: .shopping
    ),
    Feature(
        title: "Meals",
        subtitle: "Plan the week",
        systemImage: "fork.knife",
        accent: .meals,
        route: .meal
    ),
    Feature(
        title: "Calendar",
        subtitle: "Family events",
        systemImage: "calendar",
        accent: .calendar,
        route: nil
    ),
    Feature(
        title: "Birthdays",
        subtitle: "Never miss one",
        systemImage: "birthday.cake.fill",
        accent: .birthdays,
        route: .birthday
    ),
    Feature(
        title: "Wishlists",
        subtitle: "Gift ideas",
        systemImage: "gift.fill",
        accent: .wishlists,
        route: .wishlist
    ),
    Feature(
        title: "Family Map",
        subtitle: "See where everyone is",
        systemImage: "map.fill",
        accent: .map,
        route: .familyMap
    ),
]

struct HomeScreen: View {
    let viewModel: HomeViewModel
    let onOpen: (Route) -> Void
    let onOpenCalendarTab: () -> Void
    let onOpenFamily: () -> Void

    @Environment(\.colorScheme) private var colorScheme

    private let columns = [
        GridItem(.flexible(), spacing: 14),
        GridItem(.flexible(), spacing: 14),
    ]

    var body: some View {
        ScrollView {
            VStack(spacing: 14) {
                HomeHeader(state: viewModel.state, dark: colorScheme == .dark, onOpenFamily: onOpenFamily)

                if viewModel.state.isLoading {
                    LoadingState()
                } else if viewModel.state.loadError {
                    ErrorBanner(message: L("Couldn't load your data. Pull to refresh."))
                } else {
                    if viewModel.state.hasSummary {
                        summarySection
                        SectionHeader(text: L("Quick access"))
                    }
                    LazyVGrid(columns: columns, spacing: 14) {
                        ForEach(features) { feature in
                            FeatureTile(feature: feature) {
                                if let route = feature.route {
                                    onOpen(route)
                                } else {
                                    onOpenCalendarTab()
                                }
                            }
                        }
                    }
                }
            }
            .padding(.horizontal, Spacing.screenEdge)
            .padding(.vertical, Spacing.xl)
        }
        .ambientBackground()
        .refreshable { viewModel.refresh() }
        .resumeEffect { viewModel.refresh() }
    }

    @ViewBuilder private var summarySection: some View {
        let state = viewModel.state
        VStack(spacing: 9) {
            if let tonight = state.tonightMeal {
                SummaryCard(
                    systemImage: "fork.knife",
                    feature: .meals,
                    label: L("TODAY"),
                    value: tonight,
                    detail: nil
                ) { onOpen(.meal) }
            }
            if let event = state.nextEvent {
                EventSummaryCard(
                    event: event,
                    detail: state.nextEventWhen,
                    members: state.familyMembers
                ) {
                    onOpenCalendarTab()
                }
            }
            if state.shoppingRemaining > 0 {
                SummaryCard(
                    systemImage: "cart.fill",
                    feature: .shopping,
                    label: L("SHOPPING"),
                    value: L("\(state.shoppingRemaining) left to buy"),
                    detail: nil
                ) { onOpen(.shopping) }
            }
            if let birthdayName = state.nextBirthdayName {
                SummaryCard(
                    systemImage: "birthday.cake.fill",
                    feature: .birthdays,
                    label: L("NEXT BIRTHDAY"),
                    value: birthdayName,
                    detail: state.nextBirthdayWhen
                ) { onOpen(.birthday) }
            }
        }
    }
}

// MARK: - Pieces

private struct SummaryCard: View {
    let systemImage: String
    let feature: FeatureAccent
    let label: String
    let value: String
    let detail: String?
    let onTap: () -> Void

    var body: some View {
        ListCard(onTap: onTap) {
            FeatureBadge(systemImage: systemImage, feature: feature)
            Spacer().frame(width: 14)
            VStack(alignment: .leading, spacing: 2) {
                Text(label)
                    .font(.eyebrow)
                    .tracking(0.7)
                    .foregroundStyle(feature.stroke)
                Text(value)
                    .font(.cardTitle)
                    .foregroundStyle(Color.appOnSurface)
                    .lineLimit(1)
                if let detail {
                    Text(detail)
                        .font(.caption)
                        .foregroundStyle(Color.appCaption)
                        .lineLimit(1)
                }
            }
            Spacer()
            Image(systemName: "chevron.right")
                .font(.system(size: 13, weight: .semibold))
                .foregroundStyle(Color.appCaption)
        }
    }
}

/// NEXT EVENT card — mirrors the calendar event's own icon, colour and attendees so the
/// dashboard reflects the real event rather than a generic calendar glyph.
private struct EventSummaryCard: View {
    let event: CalendarEventModel
    let detail: String?
    let members: [UserModel]
    let onTap: () -> Void

    /// Creator first, then everyone tagged as "going with".
    private var people: [UserModel] {
        ([event.userId] + event.attendeeIds).compactMap { id in members.first { $0.id == id } }
    }

    private var accent: Color? {
        event.color.map { Color(hex: UInt32(truncatingIfNeeded: $0)) }
    }

    var body: some View {
        ListCard(onTap: onTap) {
            FeatureBadge(
                systemImage: IconKeyMap.calendarSymbol(event.icon),
                feature: .calendar,
                colorOverride: accent
            )
            Spacer().frame(width: 14)
            VStack(alignment: .leading, spacing: 2) {
                Text(L("NEXT EVENT"))
                    .font(.eyebrow)
                    .tracking(0.7)
                    .foregroundStyle(accent ?? FeatureAccent.calendar.stroke)
                Text(event.activity)
                    .font(.cardTitle)
                    .foregroundStyle(Color.appOnSurface)
                    .lineLimit(1)
                if let detail {
                    Text(detail)
                        .font(.caption)
                        .foregroundStyle(Color.appCaption)
                        .lineLimit(1)
                }
            }
            Spacer()
            if !people.isEmpty {
                HStack(spacing: -10) {
                    ForEach(people.prefix(3)) { person in
                        InitialAvatar(user: person, size: 36)
                            .overlay(Circle().strokeBorder(Color.appSurface, lineWidth: 2))
                    }
                }
                .padding(.leading, 4)
            }
            Image(systemName: "chevron.right")
                .font(.system(size: 13, weight: .semibold))
                .foregroundStyle(Color.appCaption)
        }
    }
}

private struct HomeHeader: View {
    let state: HomeUiState
    let dark: Bool
    let onOpenFamily: () -> Void

    var body: some View {
        VStack(spacing: Spacing.lg) {
            HStack(alignment: .center, spacing: Spacing.md) {
                VStack(alignment: .leading, spacing: 2) {
                    Text(greetingLine)
                        .font(.greeting)
                        .foregroundStyle(Color.appOnBackground)
                    Text("Here's everything your family shares.")
                        .font(.caption)
                        .foregroundStyle(Color.appOnSurfaceVariant)
                }
                .frame(maxWidth: .infinity, alignment: .leading)
                if let user = state.user {
                    InitialAvatar(user: user, size: 44)
                }
            }

            if state.user != nil {
                if let family = state.family {
                    FamilyCard(
                        family: family,
                        memberCount: state.memberCount,
                        dark: dark,
                        onTap: onOpenFamily
                    )
                } else {
                    NoFamilyBanner(onOpenFamily: onOpenFamily)
                }
            }
        }
    }

    private var greetingLine: String {
        let greeting = timeBasedGreeting(locale: appLocale)
        let firstName = state.user?.name.split(separator: " ").first.map(String.init) ?? ""
        return firstName.isEmpty ? greeting : "\(greeting), \(firstName)"
    }
}

private struct FamilyCard: View {
    let family: FamilyModel
    let memberCount: Int
    let dark: Bool
    let onTap: () -> Void

    var body: some View {
        Button(action: onTap) {
            HStack(spacing: 14) {
                ZStack {
                    Circle().fill(Color.white.opacity(0.2))
                    if let photoUrl = family.photoUrl, let url = URL(string: photoUrl) {
                        LazyImage(url: url) { phase in
                            if let image = phase.image {
                                image.resizable().scaledToFill()
                            }
                        }
                        .clipShape(Circle())
                    } else {
                        Image(systemName: "person.3.fill")
                            .font(.system(size: 22))
                            .foregroundStyle(.white)
                    }
                }
                .frame(width: 52, height: 52)

                VStack(alignment: .leading, spacing: 2) {
                    Text(family.name)
                        .font(.system(size: 18, weight: .bold))
                        .foregroundStyle(.white)
                    Text(memberCount == 1 ? L("1 member") : L("\(memberCount) members"))
                        .font(.system(size: 13))
                        .foregroundStyle(.white.opacity(0.85))
                }
                Spacer()
                Image(systemName: "chevron.right")
                    .font(.system(size: 15, weight: .semibold))
                    .foregroundStyle(.white.opacity(0.9))
            }
            .padding(Spacing.xl)
            .background(Gradients.hero(dark: dark))
            .clipShape(RoundedRectangle(cornerRadius: 24, style: .continuous))
            // Identity surface — inset shine + hairline + coloured drop shadow (spec 2a).
            .overlay(
                RoundedRectangle(cornerRadius: 24, style: .continuous)
                    .strokeBorder(Color.white.opacity(0.25), lineWidth: 0.5)
            )
            .shadow(
                color: (dark ? Color.black : Palette.indigo600).opacity(dark ? 0.4 : 0.28),
                radius: dark ? 17 : 15,
                x: 0,
                y: dark ? 12 : 10
            )
        }
        .buttonStyle(PressScaleButtonStyle())
    }
}

private struct NoFamilyBanner: View {
    let onOpenFamily: () -> Void

    var body: some View {
        HStack {
            VStack(alignment: .leading, spacing: 2) {
                Text("No family yet")
                    .font(.titleMedium)
                    .foregroundStyle(Color.appOnPrimaryContainer)
                Text("Join or create a family to get started")
                    .font(.labelMedium)
                    .foregroundStyle(Color.appOnPrimaryContainer.opacity(0.8))
            }
            .frame(maxWidth: .infinity, alignment: .leading)
            Button("Get started", action: onOpenFamily)
                .font(.bodyMedium.weight(.semibold))
                .foregroundStyle(Color.appPrimary)
        }
        .padding(Spacing.lg)
        .background(Color.appPrimaryContainer)
        .clipShape(RoundedRectangle(cornerRadius: Radius.medium, style: .continuous))
    }
}

private struct FeatureTile: View {
    let feature: Feature
    let onTap: () -> Void

    var body: some View {
        Button(action: onTap) {
            VStack(alignment: .leading, spacing: 0) {
                FeatureBadge(systemImage: feature.systemImage, feature: feature.accent, size: 36)
                Spacer(minLength: Spacing.sm)
                Text(LocalizedStringKey(feature.title))
                    .font(.cardTitle)
                    .foregroundStyle(Color.appOnSurface)
                    .lineLimit(1)
                Text(LocalizedStringKey(feature.subtitle))
                    .font(.system(size: 12))
                    .foregroundStyle(Color.appCaption)
                    .lineLimit(1)
            }
            .padding(14)
            .frame(maxWidth: .infinity, minHeight: 112, alignment: .leading)
            .glassCard(cornerRadius: Radius.row)
            // Pin the hit region to the visible tile — the glass effect inside a
            // LazyVGrid otherwise leaves the Button's tap target misaligned.
            .contentShape(RoundedRectangle(cornerRadius: Radius.row, style: .continuous))
        }
        .buttonStyle(PressScaleButtonStyle())
        .accessibilityLabel(Text("\(Text(LocalizedStringKey(feature.title))) feature"))
    }
}

File: android/app/src/main/java/com/sandnes/familyapp/ui/home/HomeScreen.kt
@file:Suppress("ktlint:standard:function-naming")

package com.sandnes.familyapp.ui.home

import android.content.res.Configuration
import androidx.annotation.StringRes
import androidx.compose.animation.core.animateFloatAsState
import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.interaction.MutableInteractionSource
import androidx.compose.foundation.interaction.collectIsPressedAsState
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.aspectRatio
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.statusBarsPadding
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.GridItemSpan
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.lazy.grid.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Cake
import androidx.compose.material.icons.filled.CalendarMonth
import androidx.compose.material.icons.filled.CardGiftcard
import androidx.compose.material.icons.filled.ChevronRight
import androidx.compose.material.icons.filled.Groups
import androidx.compose.material.icons.filled.Map
import androidx.compose.material.icons.filled.Restaurant
import androidx.compose.material.icons.filled.ShoppingCart
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.remember
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.scale
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.platform.LocalConfiguration
import androidx.compose.ui.platform.LocalDensity
import androidx.compose.ui.platform.LocalWindowInfo
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.semantics.Role
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.role
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.hilt.navigation.compose.hiltViewModel
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import coil3.compose.AsyncImage
import com.sandnes.familyapp.R
import com.sandnes.familyapp.data.CalendarEventModel
import com.sandnes.familyapp.data.UserModel
import com.sandnes.familyapp.ui.components.ErrorBanner
import com.sandnes.familyapp.ui.components.InitialAvatar
import com.sandnes.familyapp.ui.components.ListSkeleton
import com.sandnes.familyapp.ui.components.RefreshOnResume
import com.sandnes.familyapp.ui.components.SectionHeader
import com.sandnes.familyapp.ui.navigation.Routes
import com.sandnes.familyapp.ui.theme.FeatureAccent
import com.sandnes.familyapp.ui.theme.FeatureBadge
import com.sandnes.familyapp.ui.theme.IconKeyMap
import com.sandnes.familyapp.ui.theme.Indigo600
import com.sandnes.familyapp.ui.theme.Radius
import com.sandnes.familyapp.ui.theme.Spacing
import com.sandnes.familyapp.ui.theme.appDarkTheme
import com.sandnes.familyapp.ui.theme.glassCard
import com.sandnes.familyapp.ui.theme.heroGradient
import com.sandnes.familyapp.ui.theme.hexColor

private data class Feature(
    @StringRes val titleRes: Int,
    @StringRes val subtitleRes: Int,
    val icon: ImageVector,
    val accent: FeatureAccent,
    val route: String,
)

// All feature values are compile-time constants — hoisted to avoid reallocating on every recomposition.
private val features =
    listOf(
        Feature(R.string.shopping, R.string.shared_lists, Icons.Filled.ShoppingCart, FeatureAccent.Shopping, Routes.SHOPPING),
        Feature(R.string.meals, R.string.plan_the_week, Icons.Filled.Restaurant, FeatureAccent.Meals, Routes.MEAL),
        Feature(R.string.calendar, R.string.family_events, Icons.Filled.CalendarMonth, FeatureAccent.Calendar, Routes.CALENDAR),
        Feature(R.string.birthdays, R.string.never_miss_one, Icons.Filled.Cake, FeatureAccent.Birthdays, Routes.BIRTHDAY),
        Feature(R.string.wishlists, R.string.gift_ideas, Icons.Filled.CardGiftcard, FeatureAccent.Wishlists, Routes.WISHLIST),
        Feature(R.string.family_map, R.string.see_where_everyone_is, Icons.Filled.Map, FeatureAccent.Map, Routes.FAMILY_MAP),
    )

private const val PRESSED_TILE_SCALE = 0.97f
private const val MAX_ATTENDEE_AVATARS = 3

@Composable
fun HomeScreen(
    onOpen: (String) -> Unit,
    onOpenFamily: () -> Unit,
    viewModel: HomeViewModel = hiltViewModel(),
) {
    val state by viewModel.state.collectAsStateWithLifecycle()
    val dark = appDarkTheme()

    RefreshOnResume { viewModel.refresh() }

    val configuration = LocalConfiguration.current
    val windowInfo = LocalWindowInfo.current
    val density = LocalDensity.current
    val isLandscape = configuration.orientation == Configuration.ORIENTATION_LANDSCAPE
    val screenWidthDp =
        with(density) {
            windowInfo.containerSize.width
                .toDp()
                .value
        }
    val isWideWindow = screenWidthDp >= 600
    val isTablet = configuration.smallestScreenWidthDp >= 600
    // Compact width: 2 columns. Medium/Expanded (wide window): 3 columns.
    val columns = if (isWideWindow) 3 else 2
    val horizontalPadding = if (isTablet) Spacing.xxxl else Spacing.screenEdge

    val tileAspectRatio =
        when {
            isLandscape && isTablet -> 2.2f
            isLandscape -> 1.7f
            isTablet -> 1.7f
            // Mirrors the iOS FeatureTile (minHeight 112pt on a ~171pt-wide tile).
            else -> 1.6f
        }

    // Transparent so the app-level AmbientBackground shows through the glass surfaces.
    LazyVerticalGrid(
        columns = GridCells.Fixed(columns),
        modifier =
            Modifier
                .fillMaxSize()
                .statusBarsPadding(),
        contentPadding = PaddingValues(horizontal = horizontalPadding, vertical = Spacing.screenEdge),
        horizontalArrangement = Arrangement.spacedBy(Spacing.md),
        verticalArrangement = Arrangement.spacedBy(Spacing.md),
    ) {
        item(span = { GridItemSpan(maxLineSpan) }) {
            Column {
                HomeHeader(state = state, dark = dark, onOpenFamily = onOpenFamily)
                Spacer(Modifier.height(Spacing.xs))
            }
        }

        when {
            state.isLoading -> item(span = { GridItemSpan(maxLineSpan) }) { ListSkeleton(rows = 3) }
            state.loadError ->
                item(span = { GridItemSpan(maxLineSpan) }) {
                    ErrorBanner(message = stringResource(R.string.couldn_t_load_your_data_pull_to_refresh))
                }
            else -> {
                if (state.hasSummary) {
                    item(span = { GridItemSpan(maxLineSpan) }) {
                        SummarySection(state = state, onOpen = onOpen)
                    }
                }
                items(features) { feature ->
                    FeatureTile(
                        feature = feature,
                        aspectRatio = tileAspectRatio,
                        onClick = { onOpen(feature.route) },
                    )
                }
            }
        }
    }
}

@Composable
private fun SummarySection(
    state: HomeUiState,
    onOpen: (String) -> Unit,
) {
    val tonight = state.tonightMeal
    val event = state.nextEvent
    val birthday = state.nextBirthday
    // Summary detail lines are composed here so they follow the in-app locale.
    val labels =
        SummaryLabels(
            today = stringResource(R.string.today),
            tomorrow = stringResource(R.string.tomorrow),
            todayExclaim = stringResource(R.string.today_exclaim),
            inDaysFormat = stringResource(R.string.in_days_lower),
            turnsFormat = stringResource(R.string.turns_age),
        )
    val today = remember(state) { java.time.LocalDate.now() }
    Column(verticalArrangement = Arrangement.spacedBy(Spacing.sm)) {
        if (tonight != null) {
            SummaryCard(
                icon = Icons.Filled.Restaurant,
                accent = FeatureAccent.Meals,
                label = stringResource(R.string.today_header),
                value = tonight,
                detail = null,
            ) { onOpen(Routes.MEAL) }
        }
        if (event != null) {
            EventSummaryCard(
                event = event,
                detail = eventWhen(event, today, labels),
                members = state.familyMembers,
            ) { onOpen(Routes.CALENDAR) }
        }
        if (state.shoppingRemaining > 0) {
            SummaryCard(
                icon = Icons.Filled.ShoppingCart,
                accent = FeatureAccent.Shopping,
                label = stringResource(R.string.shopping_header),
                value = stringResource(R.string.count_left_to_buy, state.shoppingRemaining),
                detail = null,
            ) { onOpen(Routes.SHOPPING) }
        }
        if (birthday != null) {
            SummaryCard(
                icon = Icons.Filled.Cake,
                accent = FeatureAccent.Birthdays,
                label = stringResource(R.string.next_birthday),
                value = birthday.name,
                detail = state.nextBirthdayDate?.let { birthdayWhen(birthday, it, today, labels) },
            ) { onOpen(Routes.BIRTHDAY) }
        }
        Spacer(Modifier.height(Spacing.xs))
        SectionHeader(stringResource(R.string.quick_access))
    }
}

@Composable
private fun SummaryCard(
    icon: ImageVector,
    accent: FeatureAccent,
    label: String,
    value: String,
    detail: String?,
    onClick: () -> Unit,
) {
    Row(
        Modifier
            .fillMaxWidth()
            .glassCard(cornerRadius = Radius.row)
            .clickable(onClick = onClick)
            .padding(Spacing.cardPadding),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        FeatureBadge(icon = icon, feature = accent, size = 38.dp, cornerRadius = Radius.badge)
        Spacer(Modifier.size(Spacing.md))
        Column(Modifier.weight(1f)) {
            Text(
                label,
                style = MaterialTheme.typography.labelMedium,
                color = accent.stroke(),
                fontWeight = FontWeight.SemiBold,
            )
            Text(
                value,
                style = MaterialTheme.typography.titleMedium,
                color = MaterialTheme.colorScheme.onSurface,
                fontWeight = FontWeight.SemiBold,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis,
            )
            if (detail != null) {
                Text(
                    detail,
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis,
                )
            }
        }
        Icon(Icons.Filled.ChevronRight, null, tint = MaterialTheme.colorScheme.onSurfaceVariant)
    }
}

/** NEXT EVENT card — shows the event's own icon, colour and attendee avatars. */
@Composable
private fun EventSummaryCard(
    event: CalendarEventModel,
    detail: String?,
    members: List<UserModel>,
    onClick: () -> Unit,
) {
    val accentOverride = hexColor(event.color)
    // Creator first, then everyone tagged as "going with".
    val people =
        remember(event, members) {
            (listOf(event.userId) + event.attendeeIds)
                .distinct()
                .mapNotNull { id -> members.firstOrNull { it.id == id } }
                .take(MAX_ATTENDEE_AVATARS)
        }
    Row(
        Modifier
            .fillMaxWidth()
            .glassCard(cornerRadius = Radius.row)
            .clickable(onClick = onClick)
            .padding(Spacing.cardPadding),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        FeatureBadge(
            icon = IconKeyMap.calendar(event.icon),
            feature = FeatureAccent.Calendar,
            size = 38.dp,
            cornerRadius = Radius.badge,
            colorOverride = accentOverride,
        )
        Spacer(Modifier.size(Spacing.md))
        Column(Modifier.weight(1f)) {
            Text(
                stringResource(R.string.next_event),
                style = MaterialTheme.typography.labelMedium,
                color = accentOverride ?: FeatureAccent.Calendar.stroke(),
                fontWeight = FontWeight.SemiBold,
            )
            Text(
                event.activity,
                style = MaterialTheme.typography.titleMedium,
                color = MaterialTheme.colorScheme.onSurface,
                fontWeight = FontWeight.SemiBold,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis,
            )
            if (detail != null) {
                Text(
                    detail,
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis,
                )
            }
        }
        if (people.isNotEmpty()) {
            Row(
                horizontalArrangement = Arrangement.spacedBy((-10).dp),
                modifier = Modifier.padding(end = Spacing.sm),
            ) {
                people.forEach { person ->
                    val color =
                        if (person.avatarColor != 0) Color(person.avatarColor) else MaterialTheme.colorScheme.primary
                    Box(
                        Modifier
                            .size(34.dp)
                            .clip(CircleShape)
                            .background(MaterialTheme.colorScheme.surface),
                        contentAlignment = Alignment.Center,
                    ) {
                        InitialAvatar(name = person.name, color = color, avatarUri = person.avatarUrl, size = 30)
                    }
                }
            }
        }
        Icon(Icons.Filled.ChevronRight, null, tint = MaterialTheme.colorScheme.onSurfaceVariant)
    }
}

@Composable
private fun HomeHeader(
    state: HomeUiState,
    dark: Boolean,
    onOpenFamily: () -> Unit,
) {
    val user = state.user
    val firstName = user?.name?.substringBefore(' ').orEmpty()
    val greeting = stringResource(remember { timeBasedGreeting() })
    val avatarColor =
        if (user != null && user.avatarColor != 0) Color(user.avatarColor) else MaterialTheme.colorScheme.primary

    Column {
        Row(
            modifier = Modifier.fillMaxWidth(),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.SpaceBetween,
        ) {
            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = if (firstName.isNotBlank()) "$greeting, $firstName" else greeting,
                    style = MaterialTheme.typography.headlineMedium,
                    fontWeight = FontWeight.Bold,
                    color = MaterialTheme.colorScheme.onBackground,
                )
                Text(
                    text = stringResource(R.string.here_s_everything_your_family_shares),
                    style = MaterialTheme.typography.bodyMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                )
            }
            if (user != null) {
                Spacer(Modifier.size(Spacing.md))
                InitialAvatar(name = user.name, color = avatarColor, avatarUri = user.avatarUrl, size = 44)
            }
        }

        Spacer(Modifier.height(Spacing.lg))

        if (user != null) {
            if (state.family != null) {
                FamilyCard(
                    familyName = state.family.name,
                    memberCount = state.memberCount,
                    dark = dark,
                    photoUrl = state.family.photoUrl,
                    onClick = onOpenFamily,
                )
            } else {
                NoFamilyBanner(onOpenFamily = onOpenFamily)
            }
        }
    }
}

@Composable
private fun FamilyCard(
    familyName: String,
    memberCount: Int,
    dark: Boolean,
    photoUrl: String?,
    onClick: () -> Unit,
) {
    // Identity surface — hairline shine border + soft indigo drop shadow (mirrors iOS FamilyCard).
    Surface(
        onClick = onClick,
        modifier =
            Modifier
                .fillMaxWidth()
                .shadow(
                    elevation = 12.dp,
                    shape = RoundedCornerShape(Radius.bigCard),
                    clip = false,
                    ambientColor = (if (dark) Color.Black else Indigo600).copy(alpha = if (dark) 0.4f else 0.28f),
                    spotColor = (if (dark) Color.Black else Indigo600).copy(alpha = if (dark) 0.4f else 0.28f),
                ),
        shape = RoundedCornerShape(Radius.bigCard),
        color = MaterialTheme.colorScheme.primaryContainer,
        border = BorderStroke(0.5.dp, Color.White.copy(alpha = 0.25f)),
    ) {
        Box(
            Modifier
                .fillMaxWidth()
                .background(heroGradient(dark))
                .padding(Spacing.xl),
        ) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Box(
                    Modifier
                        .size(52.dp)
                        .clip(CircleShape)
                        .background(Color.White.copy(alpha = 0.2f)),
                    contentAlignment = Alignment.Center,
                ) {
                    if (photoUrl != null) {
                        AsyncImage(
                            model = photoUrl,
                            contentDescription = null,
                            modifier = Modifier.fillMaxSize(),
                            contentScale = ContentScale.Crop,
                        )
                    } else {
                        Icon(Icons.Filled.Groups, null, tint = Color.White, modifier = Modifier.size(28.dp))
                    }
                }
                Spacer(Modifier.size(Spacing.md))
                Column(Modifier.weight(1f)) {
                    Text(
                        text = familyName,
                        color = Color.White,
                        style = MaterialTheme.typography.titleLarge,
                        fontWeight = FontWeight.Bold,
                    )
                    Text(
                        text =
                            if (memberCount == 1) {
                                stringResource(R.string.one_member)
                            } else {
                                stringResource(R.string.member_count, memberCount)
                            },
                        color = Color.White.copy(alpha = 0.85f),
                        style = MaterialTheme.typography.bodyMedium,
                    )
                }
                Icon(Icons.Filled.ChevronRight, null, tint = Color.White.copy(alpha = 0.9f))
            }
        }
    }
}

@Composable
private fun NoFamilyBanner(onOpenFamily: () -> Unit) {
    Surface(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(Radius.medium),
        color = MaterialTheme.colorScheme.primaryContainer,
        tonalElevation = 1.dp,
    ) {
        Row(
            modifier = Modifier.padding(Spacing.lg),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.SpaceBetween,
        ) {
            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = stringResource(R.string.no_family_yet),
                    style = MaterialTheme.typography.titleSmall,
                    fontWeight = FontWeight.SemiBold,
                    color = MaterialTheme.colorScheme.onPrimaryContainer,
                )
                Text(
                    text = stringResource(R.string.join_or_create_a_family_to_get_started),
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onPrimaryContainer.copy(alpha = 0.8f),
                )
            }
            TextButton(
                onClick = onOpenFamily,
                colors = ButtonDefaults.textButtonColors(contentColor = MaterialTheme.colorScheme.primary),
            ) {
                Text(stringResource(R.string.get_started), fontWeight = FontWeight.SemiBold)
            }
        }
    }
}

@Composable
private fun FeatureTile(
    feature: Feature,
    aspectRatio: Float,
    onClick: () -> Unit,
) {
    val interactionSource = remember { MutableInteractionSource() }
    val isPressed by interactionSource.collectIsPressedAsState()
    val scale by animateFloatAsState(
        targetValue = if (isPressed) PRESSED_TILE_SCALE else 1f,
        label = "tile-press",
    )
    val featureTitle = stringResource(feature.titleRes)
    val featureContentDescription = stringResource(R.string.feature_named, featureTitle)

    Column(
        Modifier
            .fillMaxWidth()
            .aspectRatio(aspectRatio)
            .scale(scale)
            .glassCard(cornerRadius = Radius.row)
            .clickable(interactionSource = interactionSource, indication = null, onClick = onClick)
            .semantics {
                role = Role.Button
                contentDescription = featureContentDescription
            }.padding(Spacing.md),
        verticalArrangement = Arrangement.SpaceBetween,
    ) {
        FeatureBadge(icon = feature.icon, feature = feature.accent, size = 40.dp, cornerRadius = Radius.badgeLarge)
        Column(verticalArrangement = Arrangement.spacedBy(2.dp)) {
            Text(
                text = featureTitle,
                style = MaterialTheme.typography.titleMedium,
                color = MaterialTheme.colorScheme.onSurface,
                fontWeight = FontWeight.SemiBold,
                maxLines = 1,
            )
            Text(
                text = stringResource(feature.subtitleRes),
                style = MaterialTheme.typography.bodySmall,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis,
            )
        }
    }
}
